dts/binding-template.yaml

```diff
@@ -29,6 +29,11 @@ compatible: "manufacturer,device"
 #
 # If more than one binding for a compatible is found, an error is raised.
 
+# Used to identify variants known to work with the driver for 'compatible'
```

Comment:
If we go with this, this ought to explain that `variant-compatible` isn't used by the code, and is just a "syntaxified comment" in a sense. Otherwise, I'd wonder how it's used.

Think I'm fine with it as long as that's clear. Might help bring some structure.

Reply:
(Could update the comment later if/when it gets used for other stuff. As long as it's unused, it's nice to document it, to avoid confusion.)